R/object_usage_linter.R

```diff
@@ -122,9 +122,16 @@ object_usage_linter <- function(interpret_glue = TRUE, skip_with = TRUE) {
         },
         integer(1L)
       )
-
       nodes <- unclass(lintable_symbols)[matched_symbol]
-      nodes[is.na(matched_symbol)] <- list(fun_assignment)
+
+      # fallback to line based matching if no symbol is found
+      nodes[is.na(matched_symbol)] <- lapply(which(is.na(matched_symbol)), function(i) {
+        line_based_match <- xml2::xml_find_first(
+          fun_assignment,
+          glue::glue("descendant::expr[@line1 = {res$line1[i]} and @line2 = {res$line2[i]}]")
```

Comment:
maybe a good place to use `glue_data()`?

`glue_data(res[i, ], "descendant::expr[@line1 = {line1} and @line2 = {line2}]")`

or maybe `res[i, ]` has a lot of overhead (I'm always wary of creating a lot of `data.frames` / running `[.data.frame` a lot), and `glue_data(res, "... {line1[i]} ... {line2[i]}")` would be better...

Reply:
Res will be quite small because it is `nrow(lints)`, so no worries. Also, if we don't use `res[i, ]`, glue_data saves 0 bytes, displeasing my inner code-golfer 😉